q2_composition/_ancombc.py

```diff
@@ -93,59 +110,89 @@ def _ancombc(table, metadata, formula, p_adj_method, prv_cut, lib_cut,
-    if reference_levels is not None:
-        for i in reference_levels:
-            column = i.split('::')[0]
-            level_value = i.split('::')[1]
-
-            # check that reference_level columns are present in the metadata
-            ref_column = metadata.get_column(column)
-
-            # check that each chosen column contains discrete values
-            if isinstance(ref_column, NumericMetadataColumn):
-                raise TypeError('One of the `reference_levels` columns is not'
-                                ' a categorical Metadata column. Please make'
-                                ' sure that all chosen reference level columns'
-                                ' are categorical, and not numeric.'
-                                ' Non-categorical column selected:'
-                                ' %s' % column)
-
-            if level_value not in pd.unique(meta[column].values):
-                raise ValueError('Value provided in `reference_levels`'
-                                 ' parameter not found in the associated'
-                                 ' column within the metadata. Please make'
-                                 ' sure each column::value pair is present'
-                                 ' within the metadata file.'
-                                 ' \n\n'
-                                 ' column::value pair with a value that was'
-                                 ' not found: "%s"' % i)
-
-            # check that reference_level columns are also in the formula
-            if column not in formula_terms:
-                raise ValueError('`reference_levels` column "%s" was not found'
-                                 ' within the `formula` terms.' % column)
-
-            # check that IDs associated with chosen reference level(s) are
-            # present within the input table
-            level_value_idx = meta.index[meta[column] == level_value]
-            table_idx = table.index
-
-            if level_value_idx.intersection(table_idx).empty:
-                raise ValueError('Value provided in `reference_levels`'
-                                 ' parameter not associated with any IDs'
-                                 ' in the feature table. Please make sure'
-                                 ' the value(s) selected in each'
-                                 ' column::value pair are associated with'
-                                 ' IDs present in the feature table.'
-                                 ' \n\n'
-                                 ' Value not associated with any IDs in'
-                                 ' the table: "%s"' % level_value,
-                                 ' IDs not found in table:'
-                                 ' "%s"' % level_value_idx)
-
-    else:
-        reference_levels = ''
+    reference_level_columns = []
+    for i in reference_levels:
+        column = i.split('::')[0]
+        level_value = i.split('::')[1]
```

Comment:
You can replace this with `column, level_value = i.split('::')`. This will error if there is more than one '::' in the string though, so if that error is undesirable keep it this way.